avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerdeCache.java

```diff
@@ -206,10 +221,14 @@ public static boolean isSupportedForFastSerializer(Schema.Type schemaType) {
           new FastDeserializerWithAvroSpecificImpl<>(writerSchema, readerSchema));
       if (deserializer == null) {
         deserializer = fastSpecificRecordDeserializersCache.get(schemaKey);
-        CompletableFuture.supplyAsync(() -> buildSpecificDeserializer(writerSchema, readerSchema), executor)
-            .thenAccept(d -> {
-              fastSpecificRecordDeserializersCache.put(schemaKey, d);
-            });
+        if (schemaNum.get() < schemaNumLimit) {
```

Comment:
I could this piece of code is duplicate in several different functions.
Can we abstract a function to wrap up the common logic to enforce the limit check?

For example, you could pass the actual generation logic as a runnable.

Reply:
Yes, I'm working on that.